guidance/models/transformers/_transformers.py

```diff
@@ -49,81 +53,207 @@ def __init__(
             is_ptt = isinstance(transformers_tokenizer, transformers_package.PreTrainedTokenizer)
             is_ptt_fast = isinstance(
                 transformers_tokenizer, transformers_package.PreTrainedTokenizerFast
             )
             assert is_ptt or is_ptt_fast
-
         self._orig_tokenizer = transformers_tokenizer
-        special_tokens_map = {
-            id: token for token, id in transformers_tokenizer.get_added_vocab().items()
-        }
+        byte_tokens = self._byte_tokens(transformers_tokenizer)
 
-        # build out the set of byte_string tokens
-        byte_tokens = [b""] * len(transformers_tokenizer)
-        if hasattr(transformers_tokenizer, "byte_decoder"):
-            byte_decoder = transformers_tokenizer.byte_decoder
+        # Chat Template logic
+        if chat_template is None and hasattr(self._orig_tokenizer, "chat_template"):
+            chat_template = self._orig_tokenizer.chat_template
 
-            for i in range(len(transformers_tokenizer)):
-                byte_coded = bytes(
-                    [byte_decoder[c] for c in transformers_tokenizer.convert_ids_to_tokens(i)]
+        # the superclass does most of the work once we have the tokens
+        super().__init__(
+            byte_tokens,
+            chat_template,
+            None if ignore_bos_token else transformers_tokenizer.bos_token_id,
+            transformers_tokenizer.eos_token_id,
+        )
+
+    def _tokenizer(self, model: str, **kwargs) -> Union[
+        "transformers_package.PreTrainedTokenizer",
+        "transformers_package.PreTrainedTokenizerFast",
+    ]:
+        # make sure transformers is installed
+        if not has_transformers:
+            raise ImportError("Please install transformers with `pip install transformers`")
+
+        try:
+            tokenizer = transformers_package.AutoTokenizer.from_pretrained(
+                model, use_fast=False, **kwargs
+            )
+        except ImportError:
+            raise
+        except Exception as e:
+            # Fall back for other exceptions
+            warnings.warn(f"Falling back to fast tokenizer. Could not load tokenizer for model {model} due to exception {e.__class__.__name__}: {e}")
+            tokenizer = transformers_package.AutoTokenizer.from_pretrained(
+                model, use_fast=True, **kwargs
+            )  # fall back to the fast tokenizer
+
+        return tokenizer
+
+    def _byte_tokens(
+        self,
+        transformers_tokenizer: Union[
+            "transformers_package.PreTrainedTokenizer",
+            "transformers_package.PreTrainedTokenizerFast",
+        ],
+    ) -> list[bytes]:
+
+        if hasattr(transformers_tokenizer, "byte_decoder"):
+            try:
+                self._check_byte_decoder(
+                    transformers_tokenizer.byte_decoder, transformers_tokenizer
                 )
-                byte_tokens[i] = byte_coded
+            except ByteDecoderError:
+                pass
```

Comment:
Should we log a warning here?

Reply:
Added. Let me know if the language looks good.